table/format.cc

```diff
@@ -403,4 +404,17 @@ Status UncompressBlockContents(const UncompressionInfo& uncompression_info,
                                                    ioptions, allocator);
 }
 
+Status SanitizeDbHostIdProperty(Env* env, std::string& db_host_id) {
+  if (db_host_id == kDbHostId) {
+    db_host_id.clear();
+    db_host_id.resize(kMaxHostNameLen);
+    Status s = env->GetHostName(&db_host_id.at(0), kMaxHostNameLen);
```

Comment:
If GetHostName is really allowed to not include a terminating nul character, that should be documented in the public API. And best I can tell, the only other place we call this function does not account for that presumed behavior, I guess inherited from posix gethostname. Can't we do something a bit more sane, even without changing the Env function declaration?

Reply:
The current behavior provides an indication that the input buffer length is insufficient, so the caller can retry if desired. Changing the behavior would eliminate that flexibility. I would just document the API clearly, and maybe provide a wrapper inside RocksDB for simplicity.